### PR TITLE
Engineering Excellence and Dependency Injection

### PR DESCRIPTION
**Pull request by temanbrcom**
Provided bindings for the almost entire service part and partially for the core part. Applied engineering excellence requirements for the service including the server bootstrap.

### CODE REVIEW

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserImpl.java [hunk 1]
```diff
@@ -68,21 +67,22 @@ public PreprocessedInput processLines(
...
+  private void processCopybooks(CobolSourceFormat format) {
     Multimap<String, Position> copybookNames = semanticContext.getCopybooks().getDefinitions();
     if (copybookNames.isEmpty()) {
       return;
     }
     CopybookAnalysis copybookAnalyzer = createCopybookAnalyzer();
     List<CopybookSemanticContext> contexts =
-        copybookAnalyzer.analyzeCopybooks(copybookNames, semanticContext.getCopybookUsageTracker(), format);
+        copybookAnalyzer.analyzeCopybooks(
+            copybookNames, semanticContext.getCopybookUsageTracker(), format);
```
Comment (zacanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserImpl.java]: is just formatting?

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/ClientProvider.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  * Copyright (c) 2019 Broadcom.
```
[the remark this answers is not in the document]
Comment (sergiuilie) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/ClientProvider.java]: about the year @shciubrcom, it is true that is a new class but we didn't agree on the format, we will have to change later anyway. I think the commit can be approved without modifying this.

[the remark this answers is not in the document]
Comment (ishche) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/ClientProvider.java]: yes. I did approve it.

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/Completion.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019 Broadcom.
```
[the remark this answers is not in the document]
Comment (sergiuilie) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/Completion.java]: is this class from lsp4j? do they have a license?

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserImpl.java (hunk 1 above)
Comment (temanbrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserImpl.java]: yes

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/Completion.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (temanbrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/Completion.java]: No, this is our class

**Pull request merged**